File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        maxi = nums[0]   
        res = nums[0]

        for i in range(1, len(nums)):
            maxi = max(nums[i], maxi + nums[i])  
            res = max(res, maxi)

        return res

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int best = nums[0], sum = nums[0];
        for(int i = 1; i<n;i++){
            sum = max(nums[i], sum+nums[i]);
            best = max(best, sum);
        }
        return best;        
    }
};